### PR TITLE
CIR-1694 Implemented Abb-unit tests based on Acceptances tests

### PR DESCRIPTION
**Pull request by AdamCoops**

### CODE REVIEW

File: test/v1/validation/DeemedParentValidatorSpec.scala [hunk 1]
```diff
@@ -165,44 +171,37 @@ class DeemedParentValidatorSpec extends BaseSpec {
...
+          val model = deemedParentModelUkCompany.copy(isUk = false)
           leftSideError(model.validate).errorMessage shouldBe WrongDeemedParentIsUkPartnershipAndNonUKCompany(model).errorMessage
         }
 
-        "NonUK and CTUTR is supplied" in {
+        "NonUK and No SAUTR is supplied explicitly" in {
+
+          val model = deemedParentModelNonUkCompany.copy(isUk = false, sautr = Some(UTRModel("1123456789")))
+          leftSideError(model.validate).errorMessage shouldBe WrongDeemedParentIsUkPartnershipAndNonUKCompany(model).errorMessage
+        }
```
Comment (burns238) [inline test/v1/validation/DeemedParentValidatorSpec.scala:188]: The above two tests are basically the same - deemedParentModelUkCompany is just a test asset that has been coded to contain an SAUTR. If you copy it and give it the same sautr it's still basically the same object. Is this what you were intending to test?

Same below.

File: test/v1/validation/DeemedParentValidatorSpec.scala [hunk 2]
```diff
@@ -215,6 +214,26 @@ class DeemedParentValidatorSpec extends BaseSpec {
...
+
+        "LEI is to long" in {
+          val model = deemedParentModelUkPartnership.copy(legalEntityIdentifier = Some(
+            LegalEntityIdentifierModel("QWERTYUIOPASDFGHJq11QWERTYUIOPASDFGHJq11")))
+
+          leftSideError(model.validate).errorMessage shouldBe LegalEntityIdentifierCharacterError(
+            LegalEntityIdentifierModel("QWERTYUIOPASDFGHJq11QWERTYUIOPASDFGHJq11")).errorMessage
+        }
+
+        "LEI is invalid valids" in {
```
Comment (burns238) [inline test/v1/validation/DeemedParentValidatorSpec.scala:226]: What does this mean?

File: test/v1/validation/UltimateParentValidatorSpec.scala [hunk 3]
```diff
@@ -125,90 +153,77 @@ class UltimateParentValidatorSpec extends BaseSpec {
...
-            sautr = Some(UTRModel("1123456789"))
-          )
+          val model = ultimateParentModelUkCompany.copy(isUk = false,
+            ctutr = None, sautr = Some(UTRModel("1123456789")))
 
           leftSideError(model.validate).errorMessage shouldBe WrongUltimateParentIsUkPartnershipAndNonUKCompany(model).errorMessage
         }
 
-        "NonUK and CTUTR is supplied" in {
+        "NonUK and CTUTR is supplied explicitly" in {
```
Comment (burns238) [inline test/v1/validation/UltimateParentValidatorSpec.scala:204]: Same as the first comment, I'm not sure what the difference in these tests are when we're doing the explicit and by default

File: test/v1/validation/DeemedParentValidatorSpec.scala (hunk 1 above)
Comment (AdamCoops) [reply test/v1/validation/DeemedParentValidatorSpec.scala:188]: Yes. There is an assumption being made that deemedParentModelNonUkCompany will be correct in the first test, but not in the second one. If one fails, but not the other that's a huge indicator where the problem is.

File: test/v1/validation/DeemedParentValidatorSpec.scala (hunk 2 above)
Comment (AdamCoops) [reply test/v1/validation/DeemedParentValidatorSpec.scala:226]: That's a typo lol

File: test/v1/validation/UltimateParentValidatorSpec.scala (hunk 3 above)
Comment (AdamCoops) [reply test/v1/validation/UltimateParentValidatorSpec.scala:204]: See above.

File: test/v1/validation/DeemedParentValidatorSpec.scala (hunk 1 above)
Comment (burns238) [reply test/v1/validation/DeemedParentValidatorSpec.scala:188]: Do we just want to have the second test then?

**Review by burns238 — APPROVED** [review]

**Pull request merged**